framework/core/js/src/forum/components/UserSecurityPage.tsx

```diff
@@ -0,0 +1,190 @@
+  }
+
+  /**
+   * Build an item list for the user's settings controls.
+   */
+  settingsItems() {
+    const items = new ItemList<Mithril.Children>();
+
+    ['developerTokens', 'sessions'].forEach((section) => {
+      const sectionName = `${section}Items` as 'developerTokensItems' | 'sessionsItems';
+      const sectionLocale = camelCaseToSnakeCase(section);
+
+      items.add(
+        section,
+        <FieldSet className={`Security-${section}`} label={app.translator.trans(`core.forum.security.${sectionLocale}_heading`)}>
+          {this[sectionName]().toArray()}
+        </FieldSet>
+      );
+    });
+
+    return items;
+  }
+
+  /**
+   * Build an item list for the user's access accessToken settings.
+   */
+  developerTokensItems() {
+    const items = new ItemList<Mithril.Children>();
+
+    items.add(
+      'accessTokenList',
+      this.tokens === null ? (
+        <LoadingIndicator />
+      ) : (
+        <AccessTokensList
+          key={this.tokens.length}
+          type="token"
+          ondelete={(token: AccessToken) => {
+            this.tokens = this.tokens!.filter((t) => t !== token);
+            m.redraw();
+          }}
+          tokens={this.tokens.filter((token) => !token.isSessionToken())}
+          icon="fas fa-key"
+          hideTokens={false}
+        />
+      )
+    );
+
+    if (this.user!.id() === app.session.user!.id()) {
+      items.add(
+        'newAccessToken',
+        <Button
+          className="Button"
+          disabled={!app.forum.attribute<boolean>('canCreateAccessToken')}
+          onclick={() =>
+            app.modal.show(NewAccessTokenModal, {
+              onsuccess: (token: AccessToken) => {
+                this.tokens?.push(token);
+                m.redraw();
+              },
+            })
+          }
+        >
+          {app.translator.trans('core.forum.security.new_access_token_button')}
+        </Button>
+      );
+    }
+
+    return items;
+  }
+
+  /**
+   * Build an item list for the user's access accessToken settings.
+   */
+  sessionsItems() {
+    const items = new ItemList<Mithril.Children>();
+
+    items.add(
+      'sessionsList',
+      this.tokens === null ? (
+        <LoadingIndicator />
+      ) : (
+        <AccessTokensList
+          key={this.tokens.length}
+          type="session"
+          ondelete={(token: AccessToken) => {
+            this.tokens = this.tokens!.filter((t) => t !== token);
+            m.redraw();
+          }}
+          tokens={this.tokens.filter((token) => token.isSessionToken())}
+          icon="fas fa-laptop"
+          hideTokens={true}
+        />
+      )
+    );
+
+    if (this.user!.id() === app.session.user!.id()) {
+      items.add(
+        'terminateAllOtherSessions',
+        <Button
+          className="Button"
+          onclick={this.terminateAllOtherSessions.bind(this)}
+          loading={this.loading}
+          disabled={!this.tokens?.find((token) => token.isSessionToken() && !token.isCurrent())}
```

Comment:
Is it worth adding a tooltip to explain why this is disabled, like we do for the current session button?